cli/src/utils.rs

```diff
@@ -41,6 +39,38 @@ pub fn format_devices(
     Ok(())
 }
 
+/// The minimal short id size, even if we could go lower,
+/// we want to keep a lower bound for the size of the short id to keep it simple to select the value via double click.
+pub(crate) const MINIMAL_SHORT_ID_SIZE: usize = 3;
+
+pub fn get_minimal_short_id_size<'a, I>(devices: I) -> usize
+where
+    I: Iterator<Item = &'a libparsec::DeviceID> + Clone,
+{
+    use itertools::Itertools;
+
+    // If we have duplicate, we short circuit to the max value to represent the id in hex format, which is 32 char.
```

Comment:
I don't get the point of doing this:
- the default case should take care of this fine (i.e. removing this part requires no other changes in the function), so better keep things simple
- In practice this shouldn't occur so there is no need for performance optimization here (and even if it occurs, the list of devices is most likely very small so again performances are no big deal)